src/logic/safe/store/actions/processTransaction.ts

```diff
@@ -86,6 +86,7 @@ export const processTransaction = (props: ProcessTransactionArgs): ProcessTransa
     }
 
     sender.safeTxHash = tx.safeTxHash
+    sender.isExecution = props.approveAndExecute || sender.isExecution
```

Comment:
Why do we have to set `|| sender.isExecution` as a fallback and not simply use `props.approveAndExecute`? It looks like `shouldExecuteTransaction` inside `TxSender` returns a false value (at line `49`) even when executing the next Tx.

Reply:
Also it should be
```
if (props.approveAndExecute != null) {
  sender.isExecution = props.approveAndExecute
}
```

To allow passing both `false` and `true`.

Reply:
Oh, and also you're killing the second part of the check when overwriting isExecution like this.
```
    // Execute right away?
    this.isExecution =
      !txProps.delayExecution &&
      (await shouldExecuteTransaction(this.safeInstance, this.nonce, getLastTransaction(state)))
```

Reply:
And also, `approveAndExecute` is already passed to `isExecute` here: https://github.com/gnosis/safe-react/blob/dev/src/logic/safe/store/actions/processTransaction.ts#L63

Reply:
This has been removed, relying on the logic that is already there. It seems the problem actually stemmed from not using the nonce param.